apps/sms/services/js/conversation/conversation_service.js

```diff
@@ -22,26 +38,74 @@ if (!('BridgeServiceMixin' in self)) {
   ]);
 
   var ConversationService = {
+    mobileMessageClient: null,
+
     /**
      * Initializes our service using the Service mixin's initService.
      */
     init() {
       this.initService();
+
+      this.mobileMessageClient = bridge.client(
+        'mozMobileMessageShim',
+        new BroadcastChannel('mobile-message-shim-channel')
+      ).plugin(streamClient);
     },
 
     /**
      * Stream that returns everything needed to display conversations in the
      * inbox view.
-     * @param {ServiceStream.<ConversationSummary>} The stream to use in the
-     * implementation: it will be passed autoamtically by the Bridge library.
+     * @param {ServiceStream.<ConversationSummary>} serviceStream The stream to
+     * use in the implementation: it will be passed automatically by the Bridge
+     * library.
      */
-    getAllConversations(stream) {},
+    getAllConversations(serviceStream) {
+      var getThreadsStream = this.mobileMessageClient.stream('getThreads');
```

Comment:
> actually I don't think we need the cancel operation here -- but we'll need it in getAllMessages so the thinking here is useful.

Mmm, why do you think so? Should not client call abort on service stream when view is unloading (like we entered Inbox with lots of messages, navigating to Conversation, view knows that it's going to die so the client does and it should ask all active stream to be cancelled)? Otherwise our shared worker will be iterating through cursor even when user is in a different view. It's not implemented now, but do you think we won't need it?

> Althought we also need the thread request to start as early as possible... grr I don't know then :D

Mmm, I really doubt there is any significant difference if I change the order of ```Drafts.request``` and ```client.stream```. Am I missing something?